src/connectors/snowflake/trulens/connectors/snowflake/connector.py

```diff
@@ -34,30 +34,118 @@ class SnowflakeConnector(DBConnector):
 
     def __init__(
         self,
-        account: str,
-        user: str,
-        password: str,
-        database: str,
-        schema: str,
-        warehouse: str,
-        role: str,
+        account: Optional[str] = None,
+        user: Optional[str] = None,
+        password: Optional[str] = None,
+        database: Optional[str] = None,
+        schema: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        role: Optional[str] = None,
+        snowpark_session: Optional[Session] = None,
         init_server_side: bool = False,
         database_redact_keys: bool = False,
         database_prefix: Optional[str] = None,
         database_args: Optional[Dict[str, Any]] = None,
         database_check_revision: bool = True,
     ):
-        database_args = database_args or {}
+        connection_parameters = {
+            "account": account,
+            "user": user,
+            "password": password,
+            "database": database,
+            "schema": schema,
+            "warehouse": warehouse,
```

Comment:
Its okay if these are None? we could also make the keys only exist if values are none.

Reply:
so I check one of the following is true:
1. All set and `snowpark_session` is not set in lines 60-68.
2. None set and `snowpark_session` is set in lines 87-94.